core/src/main/java/org/apache/accumulo/core/client/impl/TableOperationsImpl.java

```diff
@@ -234,8 +234,19 @@ public void create(String tableName, NewTableConfiguration ntc)
     checkArgument(tableName != null, "tableName is null");
     checkArgument(ntc != null, "ntc is null");
 
-    List<ByteBuffer> args = Arrays.asList(ByteBuffer.wrap(tableName.getBytes(UTF_8)),
-        ByteBuffer.wrap(ntc.getTimeType().name().getBytes(UTF_8)));
+    List<ByteBuffer> args = new ArrayList<>();
+    args.add(ByteBuffer.wrap(tableName.getBytes(UTF_8)));
+    args.add(ByteBuffer.wrap(ntc.getTimeType().name().getBytes(UTF_8)));
+    // Check for possible initial splits to be added at table creation
+    // Always send number of initial splits to be created, even if zero. If greater than zero,
+    // add the splits to the argument List which will be used by the FATE operations.
+    int numSplits = ntc.getSplits().size();
+    args.add(ByteBuffer.wrap(String.valueOf(numSplits).getBytes(UTF_8)));
+    if (numSplits > 0) {
+      for (Text t : ntc.getSplits()) {
+        args.add(ByteBuffer.wrap(t.toString().getBytes(UTF_8)));
```

Comment:
Splits may be binary data.  Converting Text to  a String and then to a ByteBuffer could scramble the binary data (because it encodes and then decodes utf-8).  I would recommend doing `args.add(TextUtil. getByteBuffer(t))`.  Using TextUtil properly handles the edge case where Text has a byte array thats longer than its length.

Reply:
Updated.